Yank/yank.py

```diff
@@ -131,7 +131,7 @@ class Yank(object):
         'mc_displacement_sigma': 10.0 * unit.angstroms
     }
 
-    def __init__(self, store_directory, mpicomm=None, **kwargs):
+    def __init__(self, store_directory, mpicomm=None, platform=None, **kwargs):
```

Comment:
The main problem is that stuff in `kwargs` are automatically added by `YamlBuilder` as available options, so if I do it the platform name will be specified both in the command line and in the YAML file. Unless I start adding exceptions to the way we treat options, which is something I'd like to avoid.

Beside that, in general my opinion is that we make too much use of `kwargs`, and I believe that we should move away from it when possible. I think it will make the code and its documentation much clearer.